packages/services/api/src/modules/audit-logs/providers/audit-log-recorder.ts

```diff
@@ -0,0 +1,77 @@
+import { randomUUID } from 'node:crypto';
+import { Injectable, Scope } from 'graphql-modules';
+import { captureException } from '@sentry/node';
+import { type User } from '../../../shared/entities';
+import { sql as c_sql, ClickHouse } from '../../operations/providers/clickhouse-client';
+import { Logger } from '../../shared/providers/logger';
+import { auditLogSchema, type AuditLogSchemaEvent } from './audit-logs-types';
+
+type AuditLogRecordEvent = AuditLogSchemaEvent & {
+  metadata: {
+    userId: string;
+    userEmail: string;
+    organizationId: string;
+    user: User;
+  };
+};
+
+@Injectable({
+  scope: Scope.Operation,
+  global: true,
+})
+/**
+ * Responsible for recording audit log events and storing them in ClickHouse
+ */
+export class AuditLogRecorder {
+  private logger: Logger;
+
+  constructor(
+    logger: Logger,
+    private clickHouse: ClickHouse,
+  ) {
+    this.logger = logger.child({ source: 'AuditLogRecorder' });
+  }
+
+  async record(data: AuditLogRecordEvent): Promise<void> {
+    try {
+      const { eventType } = data;
+      const { organizationId, userEmail, userId } = data.metadata;
+      this.logger.debug('Creating a log audit event (data=%o)', data);
+
+      const parsedEvent = auditLogSchema.parse(data);
+      const metadata = {
+        user: data.metadata.user,
+        ...parsedEvent,
+      };
+
+      const eventMetadata = JSON.stringify(metadata);
+      const eventTime = new Date();
+      const id = randomUUID();
+      const values = [id, eventTime, userId, userEmail, organizationId, eventType, eventMetadata];
+
+      await this.clickHouse.insert({
+        query: c_sql`
+              INSERT INTO "audit_log" (
+                "id"
+                , "event_time"
+                , "user_id"
+                , "user_email"
+                , "organization_id"
+                , "event_action"
+                , "metadata"
+              )
+              FORMAT CSV`,
+        data: [values],
+        timeout: 10000,
+        queryId: 'create-audit-log',
+      });
+    } catch (error) {
+      this.logger.error('Failed to create audit log event', error);
+      captureException(error, {
+        extra: {
+          data,
+        },
+      });
```

Comment:
TODO @kamilkisiela Should we throw the error here or swallow it?

Reply:
@dotansimha said it's fine to log error

Reply:
I was thinking about this a bit more. 

So few assumptions:
- We only log management actions to the audit log, we do not log crucial flows (like schema check/publish)
- Audit log records are of lower priority compared to the actual flow

So with these assumptions, I don't think we have a strict requirement for rollbacking/failing the flows. 

@kamilkisiela @n1ru4l @TuvalSimha what do you think would be the simplest way to do that, with the assumptions above?

Reply:
so In that case I think swallowing is fine, but reporting it to use should still be done.